cluster/images/etcd/attachlease.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"os"
+	"time"
+
+	"github.com/coreos/etcd/clientv3"
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+)
+
+func main() {
+	var keysFile string
+	var etcdHost string
+	var leaseDuration time.Duration
+	flag.StringVar(&keysFile, "file", "", "The file with ttl keys")
+	flag.StringVar(&etcdHost, "etcd-addr", "", "Etcd address")
+	flag.DurationVar(&leaseDuration, "lease-duration", time.Hour, "Lease duration")
+	flag.Parse()
+
+	if keysFile == "" {
+		glog.Fatalf("--file flag is required")
+	}
+	file, err := os.Open(keysFile)
+	if err != nil {
+		glog.Fatalf("Couldn't open %s file", keysFile)
+	}
+
+	if etcdHost == "" {
+		glog.Fatalf("--etcd-addr flag is required")
+	}
+	client, err := clientv3.New(clientv3.Config{Endpoints: []string{etcdHost}})
+	if err != nil {
+		glog.Fatalf("Error while creating etcd client: %v", err)
+	}
+
+	lease, err := client.Lease.Grant(context.TODO(), int64(leaseDuration/time.Second))
+	if err != nil {
+		glog.Fatalf("Error while creating lease: %v", err)
+	}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		key := scanner.Text()
+		getResp, err := client.KV.Get(context.TODO(), key)
```

Comment:
have you tried how long will this take? i would probably have a routine pool to do the put. 100 routine can almost speed this up 100 times.